controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -406,7 +406,8 @@ func (r *StorageClassRequestReconciler) reconcileCephFilesystemSubVolumeGroup(st
 				break
 			}
 		}
-		if dataPool == nil {
+		// since data pool alone is non-nil upon init, the following covers it
+		if len(dataPool.PoolSpec.DeviceClass) == 0 {
```

Comment:
This fix is unrelated to the unit tests changes. Can we please separate this change into its own PR.

Also, I believe a better fix will be to change link 402 to `var dataPool *rookCephv1.NamedPoolSpec = nil` and have the simpler check `if dataPool == nil` here. 

My reason is that  there is no need to create an instance of the `NamedPoolSpec` struct just for the find loop to override it, or to have the more complex check

Reply:
IMO it's related, @nb-ohad:
When I'm testing a "rejected" storage profile, it has a blank device class. Let's assume for the sake of the test that it's also the default storage profile of a storage cluster. What does that tell us? That while reconciling a ceph filesystem sub volume group, it will fail to fetch a matching `cephfilesystem` with the said blank device class since we discard rejected storage profiles from it. It is at that exact point in time that I expect an error to occur.

What you're basically saying is that there can't be a state where a default storage profile of a storage cluster is set to a "rejected' one, but currently nothing prevents it

Reply:
Got it, done